### PR TITLE
.Net: Create Plan Prompt Override

### PR DESCRIPTION
**Pull request by teresaqhoang**
### Motivation and Context

This PR adds the option to override the CreatePlan prompt.

### Description

Users can pass in a callback to `HandlebarsPlannerOptions.GetCreatePlanPrompt` that returns a valid Handlebars Template string to be used in place of the default CreatePlan prompt. 

Users have the options to select any predefined partials in their own template. These helpers are pre-registered and available for reference at the time of import override.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/Planners/Planners.Handlebars/Handlebars/HandlebarsPlannerOptions.cs [hunk 1]
```diff
@@ -24,6 +24,16 @@ public sealed class HandlebarsPlannerOptions : PlannerOptions
...
     public Func<Task<string>>? GetAdditionalPromptContext { get; set; }
 
+    /// <summary>
+    /// Callback that returns an override for the CreatePlan prompt.
+    /// </summary>
+    /// <remarks>
+    /// The callback should return a valid Handlebars template string. If this is set, the planner will use this prompt instead of the default prompt.
+    /// Devs can select any partial defined in Planners.Handlebars.CreatePlanPromptPartials namespace when constructing their own prompt.
+    /// No partials are included by default; make sure to select partials such as "{{> UserGoal }}" or {{> AdditionalContext}} if needed.
+    /// </remarks>
```
Comment (markwallace-microsoft) [inline dotnet/src/Planners/Planners.Handlebars/Handlebars/HandlebarsPlannerOptions.cs:34]: Can you add a kernel syntax sample showing how to use this and add a link in the API documentation.

**Review by markwallace-microsoft — APPROVED** [review]

File: dotnet/src/Planners/Planners.Handlebars/Handlebars/HandlebarsPlannerOptions.cs [hunk 2]
```diff
@@ -24,6 +24,16 @@ public sealed class HandlebarsPlannerOptions : PlannerOptions
...
+    public Func<string>? GetCreatePlanPrompt { get; set; }
```
Comment (RogerBarreto) [inline dotnet/src/Planners/Planners.Handlebars/Handlebars/HandlebarsPlannerOptions.cs:35]: Suggest changing the name to represent its a handler that will be used as a callback.

```suggestion
    public Func<string>? CreatePlanPromptHandler { get; set; }
```

File: dotnet/src/Planners/Planners.Handlebars/Handlebars/HandlebarsPlannerOptions.cs (hunk 1 above)
Comment (teresaqhoang) [reply dotnet/src/Planners/Planners.Handlebars/Handlebars/HandlebarsPlannerOptions.cs:34]: Added `RunOverrideCreatePlanPromptSampleAsync` to Example65_HandlebarsPlanner.cs

File: dotnet/src/Planners/Planners.Handlebars/Handlebars/HandlebarsPlannerOptions.cs (hunk 2 above)
Comment (teresaqhoang) [reply dotnet/src/Planners/Planners.Handlebars/Handlebars/HandlebarsPlannerOptions.cs]: Done

File: dotnet/src/Planners/Planners.Handlebars/Handlebars/HandlebarsPlannerOptions.cs (hunk 1 above)
Comment (teresaqhoang) [reply dotnet/src/Planners/Planners.Handlebars/Handlebars/HandlebarsPlannerOptions.cs:34]: Seems like all of our API documentation is outdated, will work with @matthewbolanos to update docs for Handlebars Planner

**Review by glahaye — APPROVED** [review]

**Pull request merged**